README.md

```diff
@@ -3,6 +3,35 @@ PHP Extension for noise generation
 
 This implements various noise generation methods in one class.
 
+# Building from source
+As for now, PHP-CPP only supports linux host, the guide given here is to only build 
+this source in linux machines. You will have to install `make` in your machine in order to build this source correctly.
+
+### 1. Building PHP Source
+The first step to build this noise is first to build the PHP source from our
+[php-build-scripts](https://github.com/NetherGamesMC/php-build-scripts) source code, with the arguments `./compile -n`. 
+This is to make sure that the PHP sources will not get removed as we need it for the next step.
+
+### 2. Installing PHP-CPP sources
+After building the PHP sources, make sure you have symlinked `php` and `php-config` into `/usr/bin`, this is to make sure that
+your php binaries can be known by PHP-CPP makefile to build itself.
+
+These are the example on how to symlink php binaries.
+```
+ln -n /root/php-build-scripts/bin/php7/bin/php-config /usr/bin/php-config
+ln -n /root/php-build-scripts/bin/php7/bin/php /usr/bin/php
+```
+
+Now, this library is NOT a php-extension, but rather a dynamic linked library for your server. Clone your preferred [PHP-CPP](https://github.com/CopernicaMarketingSoftware/PHP-CPP) sources and run `make` then followed by `make install`
```

Comment:
Should probably mention to run `make install` with proper permissions, sometimes it doesn't come straight to mind that you would need sudo